### PR TITLE
[JENKINS-56770] Do not serialize semaphores

### PR DESCRIPTION
**Pull request by res0nance**
- Set min jenkins to 2.60.3 this is the first version of jenkins to require java 8
- Bump plugin pom
- Mark the map with semaphore values as transient so it does not get serialized

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -3,11 +3,11 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>3.5</version>
+    <version>3.50</version>
   </parent>
 
   <properties>
-    <jenkins.version>1.642.3</jenkins.version>
+    <jenkins.version>2.60.3</jenkins.version>
```
Comment (cashlalala) [inline pom.xml]: Not sure this is appropriate, the plugin on 1.x Jenkins master won't be able to update anymore.
However, with current setting, it allow update on 1.x master and won't affect the update on 2.x master.
So, is there any reason we must set the min version of Jenkins to 2.60 except for Java 8 ?

**Comment by res0nance** [comment]
No other reason but if we wanted to use java 8 in the plugin code this
makes sense but im ok with reverting this

**Comment by cashlalala** [comment]
Thanks for understanding, I'd like to preserve this setting until someday we must to use 2.x Jenkins API.

**Comment by res0nance** [comment]
Reverted the version change

**Pull request merged**